et/configs.ts

```diff
@@ -0,0 +1,278 @@
+import { addToSession } from "app/session"
+import { CaseEvent, ConfigSheets, Scrubbed } from "types/types"
+import { COMPOUND_KEYS } from "app/constants"
+
+let readTime: number = 0
+let englandwales: ConfigSheets
+let scotland: ConfigSheets
+
+/**
+ * Gets the parsed JSON file contents
+ * @param regionDir repo folder that contains definitions (ie, et-ccd-definitions-scotland)
+ * @param name of the JSON file (ie, CaseField)
+ * @returns parsed JSON file
+ */
+function getJson(regionDir: string, name: string) {
+  return JSON.parse(readFileSync(`${regionDir}${sep}definitions${sep}json${sep}${name}.json`).toString())
+}
+
+/** Getter for readTime */
+export function getReadTime() {
+  return readTime
+}
+
+/** Getter for englandwales */
+function getEnglandWales() {
+  return englandwales
+}
+
+/** Getter for scotland */
+function getScotland() {
+  return scotland
+}
+
+/**
+ * Mapper function for deciding where to write objects depending on CaseTypeID
+ * @param caseTypeId a caseTypeID value
+ * @returns either englandwales or scotland
+ */
+export function getConfigSheetsForCaseTypeId(caseTypeId: string) {
+  return caseTypeId.startsWith("ET_EnglandWales") ? getEnglandWales() : getScotland()
+}
+
+/**
+ * Get all defined CaseEvent IDs in englandwales and scotland configs
+ */
+export function getCaseEventIDOpts() {
+  return getUniqueByKeyAsArray([...englandwales.CaseEvent, ...scotland.CaseEvent], 'ID')
+}
+
+/**
+ * Get all currently known FieldType IDs in englandwales and scotland configs (FieldTypes that are referenced by at least one CaseField)
+ */
+export function getKnownCaseFieldTypes() {
+  return getUniqueByKeyAsArray([...englandwales.CaseField, ...scotland.CaseField], 'FieldType')
+}
+
+/**
+ * Gets (most) options for FieldTypeParameters in englandwales and scotland configs by looking at existing CaseField FieldTypeParameters and getting all scrubbed ID options
+ * TOOD: When ComplexTypes JSON support is added. Add it here
+ */
+export function getKnownCaseFieldTypeParameters() {
+  const inUse = getUniqueByKey([...englandwales.CaseField, ...scotland.CaseField], 'FieldTypeParameter')
+  const scrubbed = getUniqueByKey([...englandwales.Scrubbed, ...scotland.Scrubbed], 'ID')
+  return Object.keys({ ...inUse, ...scrubbed })
+}
+
+/**
+ * Get all defined CaseType IDs in englandwales and scotland configs
+ */
+export function getKnownCaseTypeIds() {
+  return getUniqueByKeyAsArray([...englandwales.CaseField, ...scotland.CaseField], 'CaseTypeID')
+}
+
+/**
+ * Get all defined CaseField IDs in englandwales and scotland configs
+ */
+export function getKnownCaseFieldIds() {
+  return getUniqueByKeyAsArray([...englandwales.CaseField, ...scotland.CaseField], 'ID')
+}
+
+/**
+ * Replace in-memory configs by reading in both englandwales and scotland configs from their repo folders.
+ */
+export function readInCurrentConfig() {
+  englandwales = {
+    AuthorisationCaseField: getJson(process.env.ENGWALES_DEF_DIR, "AuthorisationCaseField"),
```

Comment:
Make these strings constants and use them to reduce code duplication between EnglandWales and Scotland

Reply:
extracted out - but added some type errors that need looking into next week